### PR TITLE
Structure preparation for billing test extension

### PR DESCRIPTION
**Pull request by jsmolar**
This PR proposed structural changes to billing tests 
- moved them one layer above from the developer portal package
- divided stripe and Braintree tests into individual packages. Braintree tests are further separated into additional packages - with and without 3DS enforcement. 
- Almost all te tests should be in separate modules, to ensure clean account for every one of them
- Improvement of API and UI invoice creation

### CODE REVIEW

**Review by mganisin — DISMISSED** [review]

File: testsuite/tests/ui/billing/braintree/test_invoice_payment.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""Braintree gateway billing tests"""
+import pytest
+
+
+@pytest.fixture(scope="module", autouse=True)
+def braintree_gateway_no_sca(braintree_gateway):
+    """Ensures Braintree billing gateway with SCA disabled"""
+    braintree_gateway(verify_3ds=False)
```
Comment (JaurbanRH) [inline testsuite/tests/ui/billing/braintree/test_invoice_payment.py:8]: Unify naming conventions. Use only `sca` or only `3ds`.

File: testsuite/tests/ui/billing/braintree/test_invoice_payment.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_card(setup_card):
```
Comment (JaurbanRH) [inline testsuite/tests/ui/billing/braintree/test_invoice_payment.py:12]: I would avoid name shadowing.

File: testsuite/tests/ui/billing/braintree/verification_3ds/conftest.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+"""Conftest for card verification enforcing 3DS verification for Braintree payment gateway """
+import pytest
+
+
+@pytest.fixture(scope="module", autouse=True)
+def braintree_gateway_sca(braintree_gateway):
+    """Ensures Braintree billing gateway with SCA enabled"""
+    braintree_gateway(verify_3ds=True)
```
Comment (JaurbanRH) [inline testsuite/tests/ui/billing/braintree/verification_3ds/conftest.py:8]: `sca` vs `3ds`

File: testsuite/tests/ui/billing/braintree/verification_3ds/test_challenge_canceled.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Successful Challenge Authentication:
+Cardholder enrolled, authentication successful, and signature verification successful.
+"""
+import pytest
+from threescale_api.errors import ApiClientError
+
+
+def test_cancel_challenge(setup_card, invoice):
+    """
+    Test scenario: Cardholder enrolled, authentication successful, and signature verification successful:
```
Comment (JaurbanRH) [inline testsuite/tests/ui/billing/braintree/verification_3ds/test_challenge_canceled.py:11]: Redundant info.

**Comment by dhlavac** [comment]
@jsmolar rebase please

**Review by mganisin — APPROVED** [review]

**Review by dhlavac — APPROVED** [review]

**Pull request merged**